workers/data_refinery_workers/processors/smashing_utils.py

```diff
@@ -1064,10 +1087,17 @@ def sync_quant_files(output_path, samples: List[Sample], filtered_samples: Dict)
                     }
                     continue
                 output_file_path = output_path + sample.accession_code + "_quant.sf"
-                sample_and_computed_files.append((latest_computed_file, output_file_path))
+                sample_and_computed_files.append((sample, latest_computed_file, output_file_path))
 
             # download this set of files, this will take a few seconds that should also help the db recover
-            executor.map(download_computed_file, sample_and_computed_files)
-            num_samples += len(sample_and_computed_files)
+            for error_reason in executor.map(download_quant_file, sample_and_computed_files):
+                if error_reason is not None:
+                    filtered_samples[sample.accession_code] = {
+                        **sample.to_metadata_dict(),
+                        "reason": error_reason,
+                    }
+                    continue
```

Comment:
I don't think this code is nested where you think it is. We loop over the samples in the `sample_page` and add them to `sample_and_computed_files`. Once we've gotten them all we map over the list and run `download_computed_files` on it. In this for loop, you're looping over every error reason from those downloads, and attributing them to `sample`, which is still defined as the last sample in `sample_page`.

I think you should instead be logging from `download_quant_file`. The rest of this doesn't really need to change I don't think.

Reply:
Good catch, I was originally doing more of the processing in this loop, but I missed that I removed where I updated sample.

Reply:
Oh wait, I forgot this is actually writing to the metadata file. In that case we do need to loop over it like this, however we need to have `download_quant_file` return the sample and the error message, so you can do `for (sample, error_reason) in executor.map(download_quant_file, sample_and_computed_files):`